pycalphad/core/eqsolver.pyx

```diff
@@ -181,7 +181,7 @@ cdef _solve_and_update_if_converged(composition_sets, comps, cur_conds, problem,
             phase_idx += 1
     return result
 
-def _solve_eq_at_conditions(comps, properties, phase_records, grid, conds_keys, state_variables, verbose,
+def _solve_eq_at_conditions(comps, properties, phase_records, grid, conds_keys, state_variables, global_min, verbose,
```

Comment:
I'm thinking that this function is starting to get complicated enough that breaking it up makes sense. There's several things this is responsible for:

1. Iterating through one condition in a broadcasted set at a time
2. Either
  a. Ignoring convergence (not solving)
  b. Solving without performing any global min (no checking for adding/removing phases)
  c. Solving with global min
3. Handling failed convergence (in cases 2b and 2c)
4. Propagating results from a single set of conditions back to the properties (closing the loop opened in 1)

I propose that we slightly refactor `_solve_eq_at_conditions` by extracting out the items in 2a-c to their own functions, which only know about one set of conditions at a time. Item 3 may also make sense to refactor, but it's fine if it's left in.

This can also be used to fix a possible spurious solve at here: https://github.com/pycalphad/pycalphad/blob/651af0735d4e3a3571564a05adacb7b4ea307257/pycalphad/core/eqsolver.pyx#L317